doc/languages-frameworks/rust.section.md

```diff
@@ -642,7 +642,228 @@ buildPythonPackage rec {
 }
 ```
 
-## Setting Up `nix-shell` {#setting-up-nix-shell}
+## `buildRustCrate`: Compiling Rust crates using Nix instead of Cargo {#compiling-rust-crates-using-nix-instead-of-cargo}
```

Comment:
```suggestion
## `buildRustCrate`: Compiling Rust crates using Nix instead of Cargo {#compiling-rust-crates-using-nix-instead-of-cargo}

Note that crate2nix needs reimplements build logic from cargo. This method can fail for crates that rely on features not implemented yet i.e. workspaces.
```

Reply:
Correct me if I am wrong, but this is my last information I had about it.

Reply:
I think this amounts to "XYZ won't work if it has bugs", which you could say about any piece of software, including Nix.  Of the four restrictions on that page, one is believed-to-be-resolved and one is actually a Nix limitation:

* A crate will only have access to its own source directory during build time

How will we remember to delete this paragraph when the three remaining outstanding issues are resolved?

I think anybody considering using `crate2nix` is going to read the `crate2nix` documentation anyways.

Another alternative would be to simply remove all the `crate2nix` references since it isn't part of nixpkgs.  But `buildRustCrate` is.

Reply:
I resolved this by simply eliminating all references to `crate2nix` other than linking to its docs.  `buildRustCrate` is the only thing we should be documenting since it is the only thing that is part of nixpkgs.